src/modules/select/wallets/keepkey/pinModal.ts

```diff
@@ -0,0 +1,178 @@
+  </div>
+`
+
+const pinInputStyles = `
+  background: inherit;
+  font-size: 0.889em;
+  font-family: inherit;
+  border-width: 1px;
+  border-style: solid;
+  border-color: inherit;
+  border-radius: 4px;
+  padding-left: 0.5rem;
+  padding-right: 4.1rem;
+  color: inherit;
+  transition: background 150ms ease-in-out;
+  height: 32px;
+`
+
+const pinInputContainerStyles = `
+  margin: 15px 0;
+  display: flex;
+`
+
+const pinInput = `
+  <div style="${pinInputContainerStyles}">
+    <input
+      id="pin-input"
+      style="${pinInputStyles}"
+      placeholder="PIN"
+      type="password"
+    />
+    <div style="margin-left: 5px">
+      ${pinButton(-1, '❮', '36px', '36px')}
+    </div>
+  </div>
+`
+const pinPadContainerStyles = `
+  align-items: center;
+  display: flex;
+  flex-flow: column;
+  padding: 20px 10px;
+`
+
+export const renderPinModal = (wallet: KeepKeyHDWallet) => {
+  const pinPadHTML = `
+    <style>.bn-branding { visibility: hidden !important; }</style>
+    <div id="pin-pad" style="${pinPadContainerStyles}" class="bn-onboard-custom">
+      ${pinButtons}
+      ${pinInput}
+    </div>
+  `
+
+  const getPinInput = () =>
+    document.getElementById('pin-input') as HTMLInputElement
+
+  const handlePinPress = (value: number) => {
+    const input = getPinInput()
+    // A value of -1 signals a backspace e.g. we delete the last char from the input
+    input.value = value === -1 ? input.value.slice(0, -1) : input.value + value
+  }
+
+  const deleteWindowProperties = () => {
+    delete (window as any)[HANDLE_PIN_PRESS]
+  }
+
+  ;(window as any)[HANDLE_PIN_PRESS] = handlePinPress
+
+  // Creates a modal component which gets mounted to the body and is passed the pin html into it's slot
+  const div = document.createElement('div')
+  div.innerHTML = pinPadHTML
+  const pinModal = new Modal({
+    target: document.body,
+    props: {
+      closeModal: () => {
+        // Cancels any action that the keepkey wallet may be doing
+        wallet.cancel()
+        deleteWindowProperties()
+        pinModal.$destroy()
+      },
+      $$slots: createSlot(div),
+      $$scope: {}
+    }
+  })
+
+  // Creates a new Button component used to trigger sending the pin to Keepkey
+  const pinPadEl = document.getElementById('pin-pad')
+  if (pinPadEl) {
+    const span = document.createElement('span')
+    span.innerHTML = `Unlock`
+    new Button({
+      target: pinPadEl,
+      props: {
+        onclick: async () => {
+          const pin = getPinInput().value
+          await wallet.sendPin(pin)
+          pinModal.$destroy()
+          deleteWindowProperties()
+        },
+        $$slots: createSlot(span),
+        $$scope: {}
+      }
+    })
+  }
+}
+
+/**
+ * createSlot - creates the necessary object needed to pass
+ * arbitrary html into a component's default slot
+ * @param element The html element which is inserted into the components slot
+ */
+function createSlot(element: HTMLElement) {
```

Comment:
@aaronbarnardsound This enables us to pass slots to a component constructor. See this [Github issue](https://github.com/sveltejs/svelte/issues/2588) for additional context.